# Scripts_Pipelines
This repository is home to my personal scripts and pipelines, complete and in development. 

Disclaimer: No pipeline or script should be followed blindly. Ensure that the script and the programs used make sense for the samples you have. You may or may not utilize all steps of these pipelines- that depends entirely on your question. Understanding the pipeline and your question will save you time by not running steps you don't need.



I use a lot of Anvi'o in my pipelines, so here are some papers/resources you should read before begining to implement any of these workflows: 


Eren, A. & Esen, Özcan & Quince, Christopher & Vineis, Joseph & Morrison, Hilary & Sogin, Mitchell & Delmont, Tom. (2015). Anvi'o: An advanced analysis and visualization platformfor 'omics data. PeerJ. 3. e1319. 10.7717/peerj.1319.
http://merenlab.org/tutorials/assembly-based-metagenomics/#co-assembly
http://merenlab.org/tutorials/infant-gut/#downloading-the-pre-packaged-infant-gut-dataset



Additionally, here are some citations of other programs I use often. 

1.	Edgar, R. C. Search and clustering orders of magnitude faster than BLAST. Bioinformatics 26, 2460–2461 (2010).
2.	Bolger, A. M., Lohse, M. & Usadel, B. Trimmomatic: a flexible trimmer for Illumina sequence data. Bioinformatics 30, 2114–2120 (2014).
3.	Anvi’o: an advanced analysis and visualization platform for ‘omics data [PeerJ]. https://peerj.com/articles/1319/.
4.	Segata, N. et al. Metagenomic microbial community profiling using unique clade-specific marker genes. Nature Methods 9, 811–814 (2012).
5.	Li, D., Liu, C.-M., Luo, R., Sadakane, K. & Lam, T.-W. MEGAHIT: an ultra-fast single-node solution for large and complex metagenomics assembly via succinct de Bruijn graph. Bioinformatics 31, 1674–1676 (2015).
6.	Langmead, B. & Salzberg, S. L. Fast gapped-read alignment with Bowtie 2. Nature Methods 9, 357–359 (2012).
7.	Li, H. et al. The Sequence Alignment/Map format and SAMtools. Bioinformatics 25, 2078–2079 (2009).
8.	Fast and sensitive taxonomic classification for metagenomics with Kaiju | Nature Communications. https://www.nature.com/articles/ncomms11257.
9.	Seemann, T. Prokka: rapid prokaryotic genome annotation. Bioinformatics 30, 2068–2069 (2014).
10.	R Core Team. R: A language and environment for statistical computing. (R Foundation for Statistical Computing, 2014).
